tests/pipeline/test_engine.py

```diff
@@ -1242,6 +1249,177 @@ def test_dollar_volume(self):
+        """
+        my_asset_column = 0
+        start_date_index = 6
+        end_date_index = 10
+
+        assets = self.asset_finder.retrieve_all(self.sids)
+        my_asset = assets[my_asset_column]
+        my_asset_filter = (AssetID() != (my_asset_column + 1))
+        num_assets = len(self.sids)
+        num_days = end_date_index - start_date_index + 1
+
+        # The order of these is meant to align with the output of `linregress`.
+        outputs = ['beta', 'alpha', 'r_value', 'p_value', 'stderr']
+
+        # Our regression factor requires that its baseline asset is not
+        # filtered out, so make sure that masking out our baseline asset does
+        # not take effect. That is, a filter which filters out only our
+        # baseline asset should produce the same result as if no mask was
+        # passed at all.
+        for mask in (NotSpecified, my_asset_filter):
+            regression_factor = RollingRegressionOfReturns(
+                baseline_asset=my_asset,
+                returns_length=returns_length,
+                regression_length=regression_length,
+                mask=mask,
+            )
+            results = self.engine.run_pipeline(
+                Pipeline(
+                    columns={
+                        output: getattr(regression_factor, output)
+                        for output in outputs
+                    },
+                ),
+                self.dates[start_date_index],
+                self.dates[end_date_index],
+            )
+            output_results = {}
+            expected_output_results = {}
+            for output in outputs:
+                output_results[output] = results[output].unstack()
+                expected_output_results[output] = full_like(
+                    output_results[output], nan,
+                )
+
+            # Run a separate pipeline that calculates returns starting 2 days
+            # prior to our start date. This is because we need
+            # (regression_length - 1) extra days of returns to compute our
+            # expected regressions.
+            returns = Returns(window_length=returns_length)
+            results = self.engine.run_pipeline(
+                Pipeline(columns={'returns': returns}),
+                self.dates[start_date_index - (regression_length - 1)],
+                self.dates[end_date_index],
+            )
+            returns_results = results['returns'].unstack()
+
+            # On each day, calculate the expected regression results for Y ~ X
+            # where Y is the asset we are interested in and X is each other
+            # asset. Each regression is calculated over `regression_length`
+            # days of data.
+            for day in range(num_days):
+                my_asset_returns = returns_results.ix[
+                    day:day+regression_length, my_asset_column,
+                ]
+                for asset in range(num_assets):
```

Comment:
`iteritems` on a DataFrame will iterate over the columns.  Clearer here might be to slice once and iterate over the columns, ala:

```
for asset, returns in returns_results.iloc[day: day + regression_length].iteritems():
    ...
```